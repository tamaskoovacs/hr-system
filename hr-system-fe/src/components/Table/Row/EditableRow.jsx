import React from "react";
import './row.scss'

const EditableRow = ({ selectedRowForEdit, handleRowEdit, handleCancel, handleSave }) => {
  return (
    <>
      {Object?.entries(selectedRowForEdit)?.map(([key, value], index) => 
        <td key={index}>
          <input type="text" required name={key} value={value ?? ''} onChange={handleRowEdit}></input>
        </td>
      )}
      <td>
        <button type="button" onClick={handleSave}>Save</button>
        <button  type="button" onClick={handleCancel}>Cancel</button>
      </td>
    </>
  );
};

export default EditableRow;
