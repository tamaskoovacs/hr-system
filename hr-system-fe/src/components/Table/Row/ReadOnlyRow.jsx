import React from "react";
import './row.scss'

const ReadOnlyRow = ({ row, headers, handleEditClick, handleDeleteClick }) => {
  return (
    <>
      {Object?.entries(row)?.map(([key, value], index) => <td key={index} data-label={headers?.[index]}>{value}</td>)}
      <td>
        <button type="button" onClick={() => handleEditClick(row)}> Edit </button>
        <button type="button" onClick={() => handleDeleteClick(row?.id)}> Delete </button>
      </td>
    </>
  );
};

export default ReadOnlyRow;
