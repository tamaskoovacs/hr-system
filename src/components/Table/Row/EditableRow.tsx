import { EditableRowProps } from 'interfaces/interfaces';
import React from 'react';
import './row.scss';

const EditableRow: React.FC<EditableRowProps> = ({ selectedRowForEdit, handleRowEdit, handleCancel }) => {
  return (
    <>
      {Object?.entries(selectedRowForEdit)?.map(([key, value], index) => (
        <td key={index}>
          {key === 'id' ? (
            <>{value}</>
          ) : (
            <input
              type="text"
              required
              name={key}
              value={value ?? ''}
              onChange={handleRowEdit}
              data-testid={`edit-row-input-${index}-${selectedRowForEdit.id}`}
            ></input>
          )}
        </td>
      ))}
      <td>
        <button type="submit" data-testid={`edit-row-save-bt-${selectedRowForEdit.id}`}>
          Save
        </button>
        <button type="button" data-testid={`edit-row-cancel-bt-${selectedRowForEdit.id}`} onClick={handleCancel}>
          Cancel
        </button>
      </td>
    </>
  );
};

export default EditableRow;
