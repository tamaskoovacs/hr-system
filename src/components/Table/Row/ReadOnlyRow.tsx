import { ReadOnlyRowProps } from 'interfaces/interfaces';
import React from 'react';
import './row.scss';

const ReadOnlyRow: React.FC<ReadOnlyRowProps> = ({ row, headers, handleEditClick, handleDeleteClick }) => {
  return (
    <>
      {Object?.entries(row)?.map(([key, value], index) => (
        <td key={index} data-label={headers?.[index]}>
          {value}
        </td>
      ))}
      <td>
        <button type="button" data-testid={`read-only-row-edit-bt-${row.id}`} onClick={() => handleEditClick(row)}>
          {'Edit'}
        </button>
        <button type="button" data-testid={`read-only-row-delete-bt-${row.id}`} onClick={() => handleDeleteClick(row?.id)}>
          {'Delete'}
        </button>
      </td>
    </>
  );
};

export default ReadOnlyRow;
