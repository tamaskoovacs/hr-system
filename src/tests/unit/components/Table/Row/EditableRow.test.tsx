import React from 'react';
import { render } from '@testing-library/react';
import EditableRow from 'components/Table/Row/EditableRow';
import { tableRow } from './ReadOnlyRow.test';

describe('EditableRow component', () => {
  test('It should matches snapshot', () => {
    const view = render(<EditableRow selectedRowForEdit={tableRow} handleRowEdit={jest.fn()} handleCancel={jest.fn()} />);
    expect(view.container).toMatchSnapshot();
  });
});
