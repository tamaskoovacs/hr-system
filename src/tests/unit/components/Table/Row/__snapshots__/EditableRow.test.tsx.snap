// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableRow component It should matches snapshot 1`] = `
<div>
  <td>
    1
  </td>
  <td>
    <input
      data-testid="edit-row-input-1-1"
      name="first_name"
      required=""
      type="text"
      value="Ewen"
    />
  </td>
  <td>
    <input
      data-testid="edit-row-input-2-1"
      name="last_name"
      required=""
      type="text"
      value=""
    />
  </td>
  <td>
    <input
      data-testid="edit-row-input-3-1"
      name="email"
      required=""
      type="text"
      value="emclewd1@bbb.org"
    />
  </td>
  <td>
    <input
      data-testid="edit-row-input-4-1"
      name="date_of_birth"
      required=""
      type="text"
      value="15/12/1991"
    />
  </td>
  <td>
    <input
      data-testid="edit-row-input-5-1"
      name="industry"
      required=""
      type="text"
      value="Telecommunications Equipment"
    />
  </td>
  <td>
    <input
      data-testid="edit-row-input-6-1"
      name="salary"
      required=""
      type="text"
      value="144392.9"
    />
  </td>
  <td>
    <input
      data-testid="edit-row-input-7-1"
      name="years_of_experience"
      required=""
      type="text"
      value="2.8"
    />
  </td>
  <td>
    <button
      data-testid="edit-row-save-bt-1"
      type="submit"
    >
      Save
    </button>
    <button
      data-testid="edit-row-cancel-bt-1"
      type="button"
    >
      Cancel
    </button>
  </td>
</div>
`;

exports[`Header component It should matches snapshot 1`] = `
<div>
  <thead>
    <tr>
      <th>
        Id
      </th>
      <th>
        First Name
      </th>
      <th>
        Last Name
      </th>
      <th>
        Email
      </th>
      <th>
        Date of birth
      </th>
      <th>
        Industry
      </th>
      <th>
        Salary
      </th>
      <th>
        Years of experience
      </th>
      <th>
        Actions
      </th>
    </tr>
  </thead>
</div>
`;

exports[`ReadOnlyRow component It should matches snapshot 1`] = `
<div>
  <td
    data-label="Id"
  >
    1
  </td>
  <td
    data-label="First Name"
  >
    Ewen
  </td>
  <td
    data-label="Last Name"
  />
  <td
    data-label="Email"
  >
    emclewd1@bbb.org
  </td>
  <td
    data-label="Date of birth"
  >
    15/12/1991
  </td>
  <td
    data-label="Industry"
  >
    Telecommunications Equipment
  </td>
  <td
    data-label="Salary"
  >
    144392.9
  </td>
  <td
    data-label="Years of experience"
  >
    2.8
  </td>
  <td>
    <button
      data-testid="read-only-row-edit-bt-1"
      type="button"
    >
      Edit
    </button>
    <button
      data-testid="read-only-row-delete-bt-1"
      type="button"
    >
      Delete
    </button>
  </td>
</div>
`;
